### ISSUE TITLE
LibJS: Take a look at MiniVM - probably the fastest universal interpreter as of now

### ISSUE THREAD

**Issue by dumblob** [SerenityOS/serenity#15225]
I just read https://awesomekling.github.io/Ladybird-a-new-cross-platform-browser-project/ and saw you are not targetting JIT in which case I wonder whether you could gain some nice speedups almost for free only by using MiniVM - a universal interpreter highly optimized for speed and memory consumption written in pure C/C++.

https://github.com/FastVM/minivm (as fast as LuaJIT in many cases)
https://github.com/FastVM/minivm/network (nearly every day there is some new commits)
https://github.com/FastVM

**Comment by linusg** [comment]
https://github.com/SerenityOS/serenity/blob/master/Documentation/FAQ.md#why-dont-you-use-library-instead-of-implementing-feature-yourself

**Comment by dumblob** [comment]
Let me rephrase my question and suggestion:

>I just read https://awesomekling.github.io/Ladybird-a-new-cross-platform-browser-project/ and saw you are not targetting JIT in which case I wonder whether you could gain some nice speedups almost for free only by using MiniVM **ideas (and perhaps MiniVM code as the best starting point)** - a universal interpreter highly optimized for speed and memory consumption written in pure C/C++.
